### PR TITLE
Added Azure.Arc.Kubernetes.Defender

### LINKED ISSUE #2124: Enable a defender extension for Arc connected Kubernetes clusters
### Existing rule

_No response_

### Suggested rule

To collect and provide data plane protections of Microsoft Defender for Containers some additional daemon set and deployments needs to be deployed to the Arc connected Kubernetes cluster.

These components are installed when the defender extension `Microsoft.KubernetesConfiguration/extensions` is installed on the cluster (`Microsoft.Kubernetes/connectedClusters`) as a sub-resource.

https://github.com/Azure/Microsoft-Defender-for-Cloud/blob/8fec880c0bc183d2d0423aa7a35be1efbff54201/Pricing%20%26%20Settings/Defender%20for%20Kubernetes/Defender%20extension%20installation%20examples/azure-defender-extension-arm-template.json

### Pillar

None

### Additional context

- https://learn.microsoft.com/azure/defender-for-cloud/defender-for-containers-enable?tabs=aks-deploy-arm%2Ck8s-deploy-resource-manager%2Ck8s-verify-api%2Ck8s-remove-api%2Caks-removeprofile-api&pivots=defender-for-container-arc#use-azure-resource-manager-to-deploy-the-defender-profile
- https://learn.microsoft.com/azure/defender-for-cloud/defender-for-containers-architecture

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2124 

Added Azure.Arc.Kubernetes.Defender rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
@BernieWhite Let me know what you think about this one.

File: src/PSRule.Rules.Azure/rules/Azure.Arc.Rule.ps1 [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Licensed under the MIT License.
+
+#
+# Validation rules for Azure Arc
+#
+
+#region Rules
+
+# Synopsis: Deploy Microsoft Defender for Containers extension for Arc-enabled Kubernetes clusters.
+Rule 'Azure.Arc.Kubernetes.Defender' -Ref 'AZR-000373' -Type 'Microsoft.Kubernetes/connectedClusters' -Tag @{ release = 'Preview'; ruleSet = '2023_06'; } {
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.Arc.Rule.ps1:11]: Let's add a label for LT-1.

```suggestion
Rule 'Azure.Arc.Kubernetes.Defender' -Ref 'AZR-000373' -Type 'Microsoft.Kubernetes/connectedClusters' -Tag @{ release = 'Preview'; ruleSet = '2023_06'; 'Azure.WAF/pillar' = 'Security'; } -Labels @{ 'Azure.MCSB.v1/control' = 'LT-1' } {
```

File: docs/en/rules/Azure.Arc.Kubernetes.Defender.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  }
+}
+```
+
+## LINKS
+
+- [Security operations](https://learn.microsoft.com/azure/architecture/framework/security/security-operations)
+- [Defender for Containers architecture](https://learn.microsoft.com/azure/defender-for-cloud/defender-for-containers-architecture)
+- [Enable Microsoft Defender for Containers](https://learn.microsoft.com/azure/defender-for-cloud/defender-for-containers-enable?pivots=defender-for-container-arc)
+- [Azure deployment reference](https://learn.microsoft.com/azure/templates/microsoft.kubernetesconfiguration/extensions)
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Arc.Kubernetes.Defender.md:112]: Let's add a reference to LT-1.

```suggestion
- [LT-1: Enable threat detection capabilities](https://learn.microsoft.com/security/benchmark/azure/baselines/azure-arc-enabled-kubernetes-security-baseline#lt-1-enable-threat-detection-capabilities)
- [Azure deployment reference](https://learn.microsoft.com/azure/templates/microsoft.kubernetesconfiguration/extensions)
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Great work @BenjaminEngeset. A few suggestions otherwise all good.

**Comment by BenjaminEngeset** [comment]
@BernieWhite Great to hear and thanks for the suggestions. Adjusted.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. All good to merge.

**Pull request merged**